src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/csip/CSIPSynchronisationService.kt

```diff
@@ -99,9 +105,100 @@ class CSIPSynchronisationService(
         event.toTelemetryProperties(it.dpsCSIPId),
       )
     } ?: let {
-      // The CSIP Report should exist already - if not, then the ordering of events may be incorrect - put back on dlq
+      // The CSIP Report Mapping  should exist already - if not, then the ordering of events may be incorrect
+      telemetryClient.trackEvent(
+        "csip-scs-synchronisation-created-failed",
+        event.toTelemetryProperties() + ("reason" to "CSIP Report for CSIP SCS not mapped"),
+      )
+      throw IllegalStateException("Received CSIP_REPORTS_UPDATED for Safer Custody Screening that has never been created/mapped")
+    }
+  }
+
+  suspend fun csipFactorInserted(event: CSIPFactorEvent) {
+    val telemetry =
+      mutableMapOf(
+        "nomisCsipReportId" to event.csipReportId,
+        "offenderNo" to event.offenderIdDisplay,
+        "nomisCsipFactorId" to event.csipFactorId,
+      )
+
+    if (event.auditModuleName == "DPS_SYNCHRONISATION") {
+      telemetryClient.trackEvent("csip-factor-synchronisation-skipped", telemetry)
+      return
+    }
+
+    val nomisFactorResponse = nomisApiService.getCSIPFactor(event.csipFactorId)
+    // Get the report mapping
+    mappingApiService.findCSIPReportByNomisId(nomisCSIPReportId = event.csipReportId)
```

Comment:
We need the dps CSIPReportId first, as this is part of the path url when sending to the CSIP Service ( wasn't sure whether this should just exist with the CSIPFactor mapping, but kept separate, for now)